src/writer/creator.cpp

```diff
@@ -192,14 +192,14 @@ namespace zim
       data->handle(dirent);
     }
 
-    void Creator::addRedirection(const std::string& path, const std::string& title, const std::string& targetPath)
+    void Creator::addRedirection(const std::string& path, const std::string& title, const std::string& targetPath, const Hints& hints)
     {
       auto dirent = data->createRedirectDirent('C', path, title, 'C', targetPath);
       if (data->dirents.size()%1000 == 0){
         TPROGRESS();
       }
 
-      data->handle(dirent);
+      data->handle(dirent, hints);
```

Comment:
Another (though, maybe, hacky) solution to the problem solved by this commit ("Allow user to pass hints for redirections") would be to *not* change the API of `DirentHandler` but create and pass here a fake empty item that has only hints. The extra parameter `hints` in 

```c++
void DirentHandler::handle(Dirent* dirent, const Hints& hints, std::shared_ptr<Item> item);
```

requires defining how conflicting or inconsistent hints coming via the second argument and `item->hints()` must be dealt with.

Reply:
I agree with the potential conflict between the hints and `item->hints()` but I'm not sure about your solution.
We would have to create a new kind of "Fake" item just to pass a `hints`. And `DirentHandler` would have to add specific checks on the item itself to know if it is a really item or a fake one.

Maybe have two differents methods, `handle(Dirent* dirent, const Hints& hints)` and `handle(Dirent* dirent, std::shared_ptr<Item> item)` would fix this.
It would avoid passing hints and item, without needing to create a fake item and with a clear separation if there is a associated item or not.

Reply:
Having two separate `handle()` methods is better than passing both `hints` and `item` in the same call.